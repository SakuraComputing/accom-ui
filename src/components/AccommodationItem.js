import React from 'react';
import {withRouter} from "react-router-dom";
import ReactHtmlParser from 'react-html-parser';

export class AccommodationItem extends React.Component {

    constructor(props){
        super(props);

        this.displayRooms = this.displayRooms.bind();
    }

    displayRooms = () => {
        this.props.history.push(`/accommodation/${this.props.id}`);
    };

    list(data) {

        if(!data) {
            return;
        }

        return data.map((node, index) => {
            return <li key ={ node.id } label={ node.label }>
                {node.label}
            </li>
        })
    }

    render() {
        return (
            <div className="accommodation-container">
                <h3>{this.props.name}</h3>
                <div>
                    { ReactHtmlParser(this.props.description)}
                </div>
                {this.props.type ?
                    <div className="accomodation-type">
                        <span>Accomodation Type: {this.props.type.name}</span>
                    </div>
                : ''}
                { this.props.facilities ?
                    <div>Facilities
                        <small className="list-header">
                            {this.list(this.props.facilities)}
                        </small>
                    </div>
                : ''}
                <div className="clearfix">
                    <button className="button button--div" onClick={this.displayRooms}>See our last available rooms ></button>
                </div>
            </div>
        )
    }
};

export default withRouter(AccommodationItem);