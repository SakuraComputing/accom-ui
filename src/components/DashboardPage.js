import React from 'react';

const DashboardPage = () =>(
    <body>
        <h1>Acommodation User Interface</h1>
    </body>
);
export default DashboardPage;