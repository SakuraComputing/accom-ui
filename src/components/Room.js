import React from 'react';
import ReactHtmlParser from 'react-html-parser';
import facilityList from '../common/facilityList';
import Availability from '../../data/accommodation_availability_data';

export class Room extends React.Component {

    constructor(props){
        super(props);
    }

    getRoomsAvailable() {
        let availability = Availability["rooms"].find(room => room.id === this.props.id);
        if(availability) {
            const { available }  = availability;
            return <span className="rooms-available">Rooms available: {available}</span>
        } else {
            return <div className="room-sold-out">Sold Out</div>;
        }
    }

    getprices () {
        const { supplement_price }   = this.props;

        if(supplement_price) {
            const { price: { price }} = supplement_price;
            return <div className="price">
                <span>Today's Price: { supplement_price ? supplement_price.price.price : '' }</span>
            </div>
        } else {
            return <div className="room-sold-out">You missed it our last room sold out</div>;
        }
    }

    displayReserveButton() {
        let availability = Availability["rooms"].find(room => room.id === this.props.id);
        if(availability) {
            return <div className="clearfix">
                <button className="button button--reserve button--div">I'll Reserve</button>
            </div>
        } else {
            return '';
        }
    }

    render() {

        return (
            <div className="room-container">
                <div className="details-container">
                    <div className="details-container__row">
                        <div className="details-container__col">
                            <div className="details-container__col__header">Room Name</div>
                            {this.props.name}
                        </div>
                        <div className="details-container__col">
                            <div className="details-container__col__header">Room Type</div>
                            {this.props.room_type_description}
                        </div>
                        <div className="details-container__col">
                            <div className="details-container__col__header">Occupancy</div>
                            <li><span>{this.props.occupancy} pers. </span>(Minimum: {this.props.min_occupancy})</li>
                        </div>
                        <div className="details-container__col">
                            { this.getRoomsAvailable() }
                        </div>
                        <div className="details-container__col">
                            { this.getprices()}
                        </div>
                    </div>
                </div>
                <div className="list-header">
                    <div>Facilities</div>
                    { this.props.facilities ? facilityList(this.props.facilities) : <li>Only Basic Facilities Avaliable</li>}
                </div>
                <div className="room-desc">
                    *{ReactHtmlParser(this.props.description)}
                </div>
                {this.displayReserveButton()}
            </div>
        )
    }
};

export default Room;