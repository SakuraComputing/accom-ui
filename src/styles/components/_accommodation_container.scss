.accommodation-container {
  margin: 3% auto 0 ;
  padding: 0 $m-size;
  max-width: 100rem;
  border: 1px solid darken($off-white, 50%);
  border-radius: 5px
}

.accomodation-type {
  font-size: $font-size-small;
  background: $light-orange;
  color: $off-white;
  margin: 10px 0;
  padding: 4px;
  border-radius: 3px;
  width: fit-content;
}
