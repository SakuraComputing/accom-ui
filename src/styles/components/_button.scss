.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-medium;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
  margin-top: 5px;
  margin-bottom: 5px;
}

.button--reserve {
  margin-bottom: 5px;
}

.button--link {
  backgound: none;
}

.button--secondary {
  background: #888888;
}


.button--div {
  float: right;
}

.clearfix {
  overflow: auto;
}