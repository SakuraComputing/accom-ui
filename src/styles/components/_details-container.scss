$grid__bp-md: 90;

.details-container {
  max-width: $grid__bp-md * 1rem;
  margin: 0;
  //border: 0.1px solid darken($light-blue, 10%);
  &--fluid {
    margin: 0;
    max-width: 100%;
  }
  &__row {
    display: flex;
    flex-wrap: wrap;
    width: 100%;
    height: 90px;
    margin-top: 2%;
  }
  &__col {
    flex-basis: (100 / 5 ) * 1%;
    align-text: center;
    border: 1px solid darken($light-blue, 10%);
  }
  &__col__header {
    display: flex;
    background-color: $dark-blue;
    padding: 5%;
    width: 100%;
    color: $off-white;
  }
}



//Styling of grid for demostration purposes
//.details-container {
//  background-color: #88d8b0;
//  padding: 10px;
//  box-sizing: border-box;
//  &__row {
//    background-color: #ffeead;
//  }
//  &__col {
//    height: 30px;
//    background-color: #ff6f69;
//    outline: 	1px solid #ffcc5c;
//    margin: 10px 0;
//    display: flex;
//    align-items: center;
//    justify-content: space-around;
//  }
//}