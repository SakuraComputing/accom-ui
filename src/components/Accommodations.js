import React from 'react';
import AccommodationItem from'./AccommodationItem';
import data from '../../data/accommodation_data';
import Pagination from '../common/Pagination';

export class Accommodations extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            accommodations: [],
            renderedAccommodations: [],
            page: 1,
            number_iterations: 2,
            search: ''
        };
        this.handlePageChange = this.handlePageChange.bind(this);
        this.updateSearch = this.updateSearch.bind(this);
    }

    handlePageChange(page) {
        const renderedAccommodations = this.state.accommodations.slice((page - 1) * this.state.number_iterations, (page -1) * this.state.number_iterations + this.state.number_iterations);
        this.setState({ page, renderedAccommodations});
    }

    updateSearch(e) {
        if(e) {
            let newSearch = this.state.accommodations.filter(
                (accommodation) => {
                    return accommodation.name.includes(e.target.value);
                }
            );
            
            const page = 1;
            const renderedAccommodations = newSearch.slice((page - 1) * this.state.number_iterations, (page -1) * this.state.number_iterations + this.state.number_iterations);
            this.setState({
                newSearch,
                renderedAccommodations: newSearch.slice(0,this.state.number_iterations),
                total: newSearch.length
            });
        }
    }

    componentDidMount() {
        const accommodations = data["accommodations"];
        console.log("Where",typeof accommodations);
        this.setState({
            accommodations,
            renderedAccommodations: accommodations.slice(0,this.state.number_iterations),
            total: accommodations.length
        });
    }

    render() {

        const { page, total, renderedAccommodations } = this.state;

        return(
            <div className="accommodations">
                <div className="content-container">
                    <div className="row">
                        <div className="col-md-1this.state.number_iterations">
                            <h1 className="display-4 text-center">Accommodations</h1>
                            <input type="text"
                                className="text-input"
                                placeholder="Search Hotels"
                                onChange={this.updateSearch}
                            />
                            <ul>
                                {
                                    renderedAccommodations.map(accommodation =>
                                    <div key={accommodation.id}>
                                        <AccommodationItem {...accommodation}/>
                                    </div>
                                    )
                                }
                            </ul>
                            <Pagination
                                margin={this.state.number_iterations}
                                page={page}
                                count={Math.ceil(total / this.state.number_iterations)}
                                onPageChange={this.handlePageChange}
                            />
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default Accommodations;
