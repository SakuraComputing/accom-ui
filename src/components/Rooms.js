import React from 'react';
import { connect } from 'react-redux';
import Room from './Room';
import facilityList from '../common/facilityList';

export class Rooms extends React.Component {

    constructor(props) {
        super(props);
        this.getRooms = this.getRooms.bind(this);
    }

    getRooms() {
        const [ { rooms } ] = this.props.accommodations.filter(accommodation => accommodation.id === parseInt(this.props.match.params.id));
        let RoomItem;
        return RoomItem = rooms.map(room => (<Room key={room.id} {...room}/>))
    }

    render() {

        return (
            <div className="content-container">
                <h2>Rooms</h2>
                {this.getRooms()}
                { this.props.facilities ? facilityList(this.props.facilites) : ''}
            </div>
        )
    }
}
const mapStateToProps = (state) =>  ({
   accommodations: state.accommodations
});
export default connect(mapStateToProps, null)(Rooms);