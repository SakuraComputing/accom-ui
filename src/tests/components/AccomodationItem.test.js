import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { AccommodationItem } from '../../components/AccommodationItem';

describe('Accomodation Item  Page', () => {

    let wrapper, history;

    beforeEach(() => {
        const key = jest.fn();
        const accommodation = jest.fn();
        history = { push: jest.fn() };
        wrapper = shallow(<AccommodationItem key={key} accomodation={accommodation} history={history}/>);
    });
    it('should render accomodation item page correctly',() => {
        wrapper.setProps({ type: { id: 1, name: 'Apart'}});
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
    it('should render type if type exists', () => {
        wrapper.setProps({ facilities: [{ id: 1, label: 'bed'}]});
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
    it('should have a function called list', () => {
        it('should be a function', () => {
            expect(typeof list).toEqual('function');
        });
    });
    it('should return facilities list when list function is called', () => {
        const facilities = [ { id: '1', label: 'Towel'}];
        const ReturnData = wrapper.instance().list(facilities);
        const test = [<li key="1" label="Towel">Towel</li>];
        expect(ReturnData).toEqual(test);
    });
    it('should return nothing when no data is passed in', () => {
        const facilities = null;
        const ReturnData = wrapper.instance().list(facilities);
        expect(ReturnData).toBeNull;
    });
    it('should have a function called display rooms', () => {
        it('should be a function', () => {
            expect(typeof displayRooms).toEqual('function');
        });
    });

    it('should redirect to the rooms page when the rooms button is clicked', () => {
        wrapper.find('button').simulate('click');
        expect(history.push).toHaveBeenLastCalledWith('/rooms');
    });
});