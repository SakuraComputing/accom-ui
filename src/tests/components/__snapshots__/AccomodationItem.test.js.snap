// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accomodation Item  Page should render accomodation item page correctly 1`] = `
<div
  className="accommodation-container"
>
  <h3 />
  <div />
  <div
    className="accomodation-type"
  >
    <span>
      Accomodation Type: 
      Apart
    </span>
  </div>
  <div
    className="clearfix"
  >
    <button
      className="button button--div"
      onClick={[Function]}
    >
      See our last available rooms &gt;
    </button>
  </div>
</div>
`;

exports[`Accomodation Item  Page should render type if type exists 1`] = `
<div
  className="accommodation-container"
>
  <h3 />
  <div />
  <div>
    Facilities
    <small
      className="list-header"
    >
      <li
        key="1"
        label="bed"
      >
        bed
      </li>
    </small>
  </div>
  <div
    className="clearfix"
  >
    <button
      className="button button--div"
      onClick={[Function]}
    >
      See our last available rooms &gt;
    </button>
  </div>
</div>
`;
